src/ORM/Search/SearchContext.php

```diff
@@ -105,6 +105,9 @@ class SearchContext
      */
     public function __construct($modelClass, $fields = null, $filters = null)
     {
+        if ($fields !== null && !is_a($fields, FieldList::class)) {
+            throw new InvalidArgumentException('$fields must be a ' . FieldList::class . ' or null');
+        }
```

Comment:
This is a breaking change - right now I can create a new class that has all the right methods on it, and it'll work. This would break that.